tests/components/homeassistant_yellow/test_init.py

```diff
@@ -0,0 +1,96 @@
+"""Test the Home Assistant Yellow integration."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.homeassistant_yellow.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, MockModule, mock_integration
+
+
+@pytest.fixture(autouse=True)
+def mock_zha():
+    """Mock the zha integration."""
+    with patch(
+        "homeassistant.components.zha.async_setup_entry",
+        return_value=True,
+    ):
+        yield
+
+
+async def test_setup_entry(hass: HomeAssistant) -> None:
+    """Test setup of a config entry, including setup of zha."""
+    mock_integration(hass, MockModule("hassio"))
+
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={},
+        title="Home Assistant Yellow",
+    )
+    config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.homeassistant_yellow.get_os_info",
+        return_value={"board": "yellow"},
+    ) as mock_get_os_info:
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert len(mock_get_os_info.mock_calls) == 1
+
+    zha_config_entry = hass.config_entries.async_entries("zha")[0]
+    assert zha_config_entry.data == {
+        "device": {
+            "baudrate": 115200,
+            "flow_control": "hardware",
+            "path": "/dev/ttyAMA1",
+        },
+        "radio_type": "ezsp",
+    }
+    assert zha_config_entry.options == {}
+    assert zha_config_entry.title == "/dev/ttyAMA1"
```

Comment:
Maybe this is too detailed, should we just check that a zha config entry is created:
```suggestion
    assert len(hass.config_entries.async_entries("zha") == 1)
```

Reply:
Simplified